Source/io.cpp

```diff
@@ -55,9 +55,9 @@ void Write(const Eigen::MatrixXf &data, double sampling_rate,
   int channel_count = data.rows();
   int frame_count = data.cols();
 
-  auto writer = std::unique_ptr<AudioFormatWriter>(
-      format.createWriterFor(output_file.createOutputStream(), sampling_rate,
-                             channel_count, 16, StringPairArray(), 0));
+  auto writer = std::unique_ptr<AudioFormatWriter>(format.createWriterFor(new FileOutputStream(output_file), sampling_rate,

```

Comment:
To avoid the memory leak, I would do something like
```
FileOutputStream stream(output_file);
auto writer = std::unique_ptr<AudioFormatWriter>(
      format.createWriterFor(&stream, sampling_rate, channel_count, 16, StringPairArray(), 0));
```
or even 
```
auto stream = output_file.createOutputStream();
auto writer = std::unique_ptr<AudioFormatWriter>(
      format.createWriterFor(stream.get(), sampling_rate, channel_count, 16, StringPairArray(), 0));
```

Reply:
Hi! Thank you for the comment :)

Since AudioFormatWriter owns OutStream's object and delete it when AudioFormatWriter is destructed.

https://docs.juce.com/master/classAudioFormatWriter.html#ad13516434d025d9427fab9ac9b208a77:~:text=%E2%97%86-,AudioFormatWriter,-()%20%5B1/2

https://github.com/juce-framework/JUCE/blob/2b16c1b94c90d0db3072f6dc9da481a9484d0435/modules/juce_audio_formats/format/juce_AudioFormatWriter.cpp#L61

Reply:
You are right !